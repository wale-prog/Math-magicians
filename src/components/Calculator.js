import { Component } from 'react';
import calculate from './calculate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      next: '',
      operation: '',
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (event) => {
    const btnValue = event.target.value;
    this.setState((prevState) => ({
      ...calculate(prevState, btnValue),
    }));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="main-container">
        <div className="input">
          <span>
            {' '}
            {next || operation || total || '0'}
            {' '}
          </span>
        </div>
        <form className="calc-btns">
          <input type="button" value="AC" onClick={this.handleClick} />
          <input type="button" value="+/-" onClick={this.handleClick} />
          <input type="button" value="%" onClick={this.handleClick} />
          <input type="button" value="รท" className="peach" onClick={this.handleClick} />
          <input type="button" value="7" onClick={this.handleClick} />
          <input type="button" value="8" onClick={this.handleClick} />
          <input type="button" value="9" onClick={this.handleClick} />
          <input type="button" value="x" className="peach" onClick={this.handleClick} />
          <input type="button" value="4" onClick={this.handleClick} />
          <input type="button" value="5" onClick={this.handleClick} />
          <input type="button" value="6" onClick={this.handleClick} />
          <input type="button" value="-" className="peach" onClick={this.handleClick} />
          <input type="button" value="3" onClick={this.handleClick} />
          <input type="button" value="2" onClick={this.handleClick} />
          <input type="button" value="1" onClick={this.handleClick} />
          <input type="button" value="+" className="peach" onClick={this.handleClick} />
          <input type="button" value="0" className="zero" onClick={this.handleClick} />
          <input type="button" value="." onClick={this.handleClick} />
          <input type="button" value="=" className="peach" onClick={this.handleClick} />
        </form>
      </div>
    );
  }
}

export default Calculator;
