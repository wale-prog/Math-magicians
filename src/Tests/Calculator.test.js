import React from 'react';
import Calculator from '../components/Calculator';
import renderer from 'react-test-renderer'

describe('Testing the calculator', () => {
    it('Calculator is rendered', () => {
      const calc = renderer
        .create(<Calculator />)
        .toJSON();
      expect(calc).toMatchSnapshot();
    });
  });