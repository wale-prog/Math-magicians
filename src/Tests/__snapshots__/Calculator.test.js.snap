// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the calculator Calculator is rendered 1`] = `
<div
  className="calc-container"
>
  <h2>
    Lets do some Math!
  </h2>
  <div
    className="main-container"
  >
    <div
      className="input"
    >
      <span>
        0
      </span>
    </div>
    <form
      className="calc-btns"
    >
      <input
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="peach"
        onClick={[Function]}
        type="button"
        value="รท"
      />
      <input
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="peach"
        onClick={[Function]}
        type="button"
        value="x"
      />
      <input
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="peach"
        onClick={[Function]}
        type="button"
        value="-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        className="peach"
        onClick={[Function]}
        type="button"
        value="+"
      />
      <input
        className="zero"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="peach"
        onClick={[Function]}
        type="button"
        value="="
      />
    </form>
  </div>
</div>
`;
